base/cluster_n1ql.go

```diff
@@ -181,9 +181,11 @@ func (cl *ClusterOnlyN1QLStore) runQuery(statement string, n1qlOptions *gocb.Que
 	if n1qlOptions == nil {
 		n1qlOptions = &gocb.QueryOptions{}
 	}
-	queryResults, err := cl.cluster.Query(statement, n1qlOptions)
-
-	return queryResults, err
+	if !cl.supportsCollections {
+		return cl.cluster.Query(statement, n1qlOptions)
+	}
+	scope := cl.cluster.Bucket(cl.bucketName).Scope(cl.scopeName)
```

Comment:
I think opening a new bucket connection here defeats the purpose of the cluster-based connection, if it's opening a connection to the bucket under the hood.

Reply:
I've ignored this for now.